services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/ResumableFileDownload.java

```diff
@@ -125,6 +143,129 @@ public Optional<Long> totalSizeInBytes() {
         return Optional.ofNullable(totalSizeInBytes);
     }
 
+    /**
+     * Persists this download object to a file in Base64-encoded JSON format.
+     *
+     * @param path The path to the file to which you want to write the serialized download object.
+     * @throws IOException If the data cannot be persisted
+     */
+    public void writeToFile(Path path) throws IOException {
```

Comment:
Shouldn't be throwing checked exceptions since this is a public interface

Reply:
ok, thanks

Reply:
Updated all APIs with checked exceptions to throw SdkClientException with a relevant message.